e2e/helpers.ts

```diff
@@ -776,6 +777,90 @@ export async function transactionStatus() {
   return txnStatus;
 }
 
+export async function waitForAndCheckTransaction(
+  provider: providers.Provider,
+  txHash: string,
+  maxAttempts = 20,
+): Promise<{
+  status: 'success' | 'failure' | 'timeout';
+  receipt: providers.TransactionReceipt | null;
+}> {
+  let attempts = 0;
+  while (attempts < maxAttempts) {
+    try {
+      console.log(`Checking transaction status, attempt ${attempts + 1}`);
+      const txnReceipt = await provider.getTransactionReceipt(txHash);
```

Comment:
doesn't provider.waitForTransaction work here?

Reply:
yeah lemme modify it a lil

Reply:
hmm. i couldn't get it to work the way i tried. I don't mind the custom helper tbh if it works and includes some other useful logic. Do you see a big reason to use this method over the one i used?

Reply:
not really just because it should do what your doing here, but if it didn't work that's fine